scripts/cleanup_openstack_instance_snapshot.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Cleanup unused instance snapshot from glance repository
+
+Usage: scripts/cleanup_openstack_instance_snapshot.py [optional list of provider keys]
+
+If no providers specified, it will cleanup all of them.
+
+"""
+import sys
+from datetime import datetime
+from datetime import timedelta
+
+import iso8601
+import tzlocal
+
+from cfme.utils.log import add_stdout_handler
+from cfme.utils.log import logger
+from cfme.utils.providers import get_mgmt
+from cfme.utils.providers import list_provider_keys
+
+LOCAL_TZ = tzlocal.get_localzone()
+GRACE_TIME = timedelta(hours=2)
+TIME_LIMIT = datetime.now(tz=LOCAL_TZ) - GRACE_TIME
+
+# log to stdout too
+add_stdout_handler(logger)
+
+
+def main(*providers):
+    """ Cleanup all snapshots name starting with test_snapshot and created by >= 2 hours before
+
+    :param providers: Lists provider keys
+    :return:
+    """
+
+    for provider_key in providers:
+        logger.info("Cleaning up {}".format(provider_key))
+        provider_obj = get_mgmt(provider_key)
+
+        try:
+            images = provider_obj.list_templates()
```

Comment:
No need to capture exception it will handle by `wrapanapi`. also if you handle an exception you will get an error at `L48` use object before initialization (`images`)

Reply:
we need this exception to continue cleanup on multiple provider such that script wont exited in case if one provider is not reachable, 
further at L48 will never execute if it goes in exception and it again continue to the for loop and check for other provider.